File: samples/typescript/messaging/broadcast_message.ts
import { ChatAnthropic } from "@langchain/anthropic";

const llm = new ChatAnthropic({ model: "claude-haiku-4-5-20251001" });

async function publisher(): Promise<string> {
  const res = await llm.invoke("Generate a one-sentence news headline about technology.");
  return res.content as string;
}

async function subscriber(role: string, topic: string): Promise<string> {
  const res = await llm.invoke(
    `You are a ${role}. React briefly to this headline: "${topic}"`
  );
  return res.content as string;
}

async function main() {
  const topic = await publisher();
  console.log("Publisher broadcast:", topic);

  const roles = ["skeptical journalist", "excited investor", "concerned ethicist"];
  const reactions = await Promise.all(
    roles.map((role) => subscriber(role, topic))
  );

  reactions.forEach((reaction, i) => {
    console.log(`\n[${roles[i]}]:`, reaction);
  });
}

main().catch(console.error);

File: samples/python/messaging/broadcast_message.py
from langchain_anthropic import ChatAnthropic

llm = ChatAnthropic(model="claude-haiku-4-5-20251001")

SUBSCRIBER_ROLES = [
    ("Optimist",  "React with enthusiasm and highlight the positive aspects."),
    ("Skeptic",   "React with critical questions and point out potential issues."),
    ("Historian", "React by connecting this to a historical parallel or precedent."),
]


def publisher() -> str:
    """Publisher agent generates a broadcast topic sentence."""
    prompt = "Write one sentence announcing a major fictional technology breakthrough."
    topic = llm.invoke(prompt).content
    return topic


def subscriber(role: str, instruction: str, message: str) -> str:
    """Subscriber agent reacts to the broadcast message according to its role."""
    prompt = (
        f"You are a {role}. {instruction}\n"
        f"Respond in 1-2 sentences to this announcement: {message}"
    )
    return llm.invoke(prompt).content


def main():
    print("=== Broadcast Message Pattern ===\n")

    topic = publisher()
    print(f"[Publisher] Broadcast: {topic}\n")

    reactions = [
        (role, subscriber(role, instruction, topic))
        for role, instruction in SUBSCRIBER_ROLES
    ]

    print("--- Subscriber Reactions ---")
    for role, reaction in reactions:
        print(f"[{role}]: {reaction}\n")


if __name__ == "__main__":
    main()
